app/src/main/java/com/csc/rssreader/ListArticlesActivity.java

```
@@ -0,0 +1,108 @@
+
+public class ListArticlesActivity extends AppCompatActivity {
+
+    private String url;
+    private ListView listViewArticles;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_articles);
+        url = getIntent().getStringExtra(GlobalContext.KEY_URL);
+        if (url == null || url.isEmpty()) {
+            url = getString(R.string.default_rss);
+        }
+        listViewArticles = (ListView) findViewById(R.id.list_articles);
+        new RSSLoadTask(this).execute();
+    }
+
+    private static class RSSLoadTask extends AsyncTask<Void, Void, Void> {
+
+        private WeakReference<ListArticlesActivity> listArticlesActivityRef;
+        ArrayList<String> dates = new ArrayList<>();
+        ArrayList<String> titles = new ArrayList<>();
+        ArrayList<String> descriptions = new ArrayList<>();
+        public RSSLoadTask(ListArticlesActivity activity) {
+            listArticlesActivityRef= new WeakReference<ListArticlesActivity>(activity);
+        }
+        @Override
+        protected Void doInBackground(Void... params) {
+            ListArticlesActivity activity = listArticlesActivityRef.get();
+            if (activity == null) {
+                return null;
+            }
+            Cursor cursor = null;
+            try {
+                cursor = activity.getContentResolver().query(Uri.parse("content://com.csc.rssreader.rssloader"), null, activity.url, null, null);
+                if (cursor != null) {
+                    while (cursor.moveToNext()) {
+                        dates.add(cursor.getString(RSSContentProvider.COLUMN_DATE));
+                        titles.add(cursor.getString(RSSContentProvider.COLUMN_TITLE));
+                        descriptions.add(cursor.getString(RSSContentProvider.COLUMN_DESCRIPTION));
+                    }
+                }
+            } catch (Exception ex) {
```

you should not ignore exception